### ISSUE TITLE
Support new OSs - Fedora 41 - Add support in QA procedures

### ISSUE THREAD

**Issue by MarcelKemp** [wazuh/wazuh#26739]
# Description

| Related issue | Epic issue |
|---|---|
| #26715 | #26723 |

| Agent tier | Central components support | OS type |
|-|-|-|
| 1 | No | Major |

# Plan

**QA**

## Considerations

- Testing environment: Deploy the new OS taking into account the following notes:
  - If the Wazuh central components are supported: 1 VM
  - If the Wazuh agent is supported: 1 VM for each architecture supported.
  - If the Wazuh central components or the Wazuh agent is not supported, deploy a Debian 12 VM for the non-supported component.
- All testing tasks must be completed for each OS architecture supported.
- The following tasks should be completed in order.

## Tasks

- [ ] Test the Wazuh dashboard one-liner deployment.
  - [ ] Deploy a Wazuh agent using the Wazuh dashboard one-liner feature provided in the Wazuh user interface with the following cases.
    - [ ] Only IP address.
    - [ ] Only FQDN.
    - [ ] IP address, agent name, and group.
- [ ] Add/Update/Check CI
  - [ ] Test the JobFlow testing tool for that specific system.
  - [ ] Add the OS and each architecture to the JobFlow testing tool.
  - [ ] Add the OS and each architecture to the GitHub Deployability and Upgrade release templates.
- [ ] Add the OS and its supported architectures to the E2E UX Tests spreadsheet (OS sheet).
- [ ] Add the OS as supported in the compatibility system spreadsheet.

**Comment by rauldpm** [comment]
## Update report

- Found an error trying to run the DTT test for the Fedora agent

```
[2025-01-20 20:06:41,418] [ERROR] PROVISION: Error while provisioning "wazuh-manager": 'NoneType' object is not iterable
[2025-01-20 20:06:41,419] [DEBUG] PROVISION: Error traceback:
Traceback (most recent call last):
  File "/home/raul/Documents/Wazuh/Repositories/wazuh-qa-automation/26723/env/lib/python3.10/site-packages/provision/provision.py", line 50, in run
    self.__provision(component)
  File "/home/raul/Documents/Wazuh/Repositories/wazuh-qa-automation/26723/env/lib/python3.10/site-packages/provision/provision.py", line 95, in __provision
    self.update_status(status)
  File "/home/raul/Documents/Wazuh/Repositories/wazuh-qa-automation/26723/env/lib/python3.10/site-packages/provision/provision.py", line 84, in update_status
    self.summary.update(status.stats)
TypeError: 'NoneType' object is not iterable
```

- Probably related to not fetching the ansible `OS family` of the remote instance, which is being referenced as `None`

```
[2025-01-20 21:38:57,739] [DEBUG] PROVISION: Get OS family for ec2-54-89-212-189.compute-1.amazonaws.com.
[2025-01-20 21:38:57,740] [DEBUG] PROVISION: Using inventory: {'all': {'hosts': {'ec2-54-89-212-189.compute-1.amazonaws.com': {'ansible_port': 2200, 'ansible_user': 'ubuntu', 'ansible_ssh_private_key_file': '/tmp/wazuh-qa/wqa-26734-ubuntu-22.04-2381/wqa-26734-ubuntu-22.04-key-9725'}}}}
[2025-01-20 21:38:57,740] [DEBUG] PROVISION: Running playbook: {'hosts': 'ec2-54-89-212-189.compute-1.amazonaws.com', 'become': True, 'gather_facts': True, 'tasks': [{'name': 'Capture ansible_os_family', 'set_fact': {'ansible_os_family': "{{ ansible_facts['distribution_file_variety'] }}", 'cacheable': 'yes'}}]}
[2025-01-20 21:38:57,869] [DEBUG] PROVISION: Playbook {'hosts': 'ec2-54-89-212-189.compute-1.amazonaws.com', 'become': True, 'gather_facts': True, 'tasks': [{'name': 'Capture ansible_os_family', 'set_fact': {'ansible_os_family': "{{ ansible_facts['distribution_file_variety'] }}", 'cacheable': 'yes'}}]} finished with status None
[2025-01-20 21:38:57,870] [DEBUG] PROVISION: OS family: None.
```

- Testing manually the ansible data fetch I got the following error

```
╰─➤  ansible all -m setup -i "host," --user=xxx --private-key=/tmp/wazuh-qa/wazuh-26723-ubuntu-22.04-5905/wazuh-26723-ubuntu-22.04-key-6364 
ERROR: Ansible could not initialize the preferred locale: unsupported locale setting
```

- Exporting the following locale data fixed the problem, apparently, this is related to an incompatibility between the host and the remote instance

```
export LC_ALL=en_US.UTF-8
export LANG=en_US.UTF-8
``` 

- Once fixed, the DTT test finished with errors 🔴 
  - YAML: [test-fedora.yaml.txt](https://github.com/user-attachments/files/18482776/test-fedora.yaml.txt)
  - LOG:
    - [jobflow.log](https://github.com/user-attachments/files/18482771/jobflow.log)
    - [testing.log](https://github.com/user-attachments/files/18482772/testing.log)

**Comment by rauldpm** [comment]
## Update report

- A new execution finished successfully 🟢 
  - YAML: [test-fedora.yaml.txt](https://github.com/user-attachments/files/18492412/test-fedora.yaml.txt)
  - LOG: [jobflow.log](https://github.com/user-attachments/files/18492399/jobflow.log)
- The previous error seems to be related to a flaky test

**Comment by rauldpm** [comment]
Blocking this issue as we need ARM resources

**Comment by rauldpm** [comment]
Unblocking as support for Fedora 41 ARM has been added https://github.com/wazuh/wazuh-automation/pull/2074

**Comment by rauldpm** [comment]
## Update report ARM

### Test the JobFlow testing tool for that specific system. 🟢 

- YAML: [Test-agent-fedora41-arm.yaml.txt](https://github.com/user-attachments/files/18591482/Test-agent-fedora41-arm.yaml.txt)
- Log: 
  - [jobflow.log](https://github.com/user-attachments/files/18591490/jobflow.log)
  - [testing.log](https://github.com/user-attachments/files/18591523/testing.log)
  - New issue https://github.com/wazuh/wazuh/issues/27760 (reported in the AMD test)
  - New issue https://github.com/wazuh/wazuh-qa-automation/issues/822

> [!NOTE]
> Although an error was found, the DTT functionality to use Fedora 41 is valid

### Test the Wazuh dashboard one-liner deployment. ⚙ 

<details><summary>Only IP address.</summary>



```

```


</details>

<details><summary>Only FQDN.</summary>


```

```


</details>

<details><summary>IP address, agent name, and group.</summary>

```

```


```

```


</details>

**Comment by hossam1522** [comment]
Please hide the public IPs and delete the edit history in the ARM part.

**Comment by rauldpm** [comment]
Those public IPs are temporal and do not exist after the test execution

**Comment by hossam1522** [comment]
LGTM!
